modules/@apostrophecms/user/index.js

```diff
@@ -440,10 +453,20 @@ module.exports = {
         if (!safeUser) {
           throw new Error('No such user in the safe.');
         }
-
-        const isVerified = await self.pw.verify(migrate(safeUser[secret + 'Hash']), attempt);
+        const key = secret + 'Hash';
+        const isVerified = await self.pw.verify(migrate(safeUser[key]), attempt);
 
         if (isVerified) {
+          if ((typeof isVerified) === 'string') {
```

Comment:
Returns a string rather than `true` if an old hash was updated to the new format.